### PR TITLE
Subtract background from all runs on loading in EngDiff UI

### LINKED ISSUE #31689: Allow user to subtract background from all runs automatically in EngDiff UI
As a user I want a button to subtract background from all runs (i.e. check the bg sub checkbox for all rows in table)

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work.**
Subtract background automatically from all runs on loading - requested behaviour from instrument scientist. It's not possible to just change default value of bgsub checkbox to be True because the various dicts in model haven't been created. Instead, the bgsub checkboxes of runs just loaded are checked to be True after the table ahs first been populated (behaves same as if the user has gone through and checked them all by hand).  To ensure the table is repopulated with the correct bgsub status I have added a setter to the model.

**To test:**
(1) In the main workbench settings, set the peak function to a `Gaussian` (File > Settings >Fitting > Default peak shape)
(2) Open the EngDiff UI (Interfaces > Diffraction > Engineering Diffraction)
(3) Go to the fitting tab, load in some focussed data - you will need two runs (to make focussed data follow the instructions in Test 1 here https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html or get the data from PR #29726)
(4) Load in one run - note the bgsub checkbox should be checked
(5) Uncheck the bgsub checkbox
(6) Load in another run - note the bg has been subtracte dform new run but the checkbox status of old run is not overwritten.

Fixes #31689 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_presenter.py [hunk 1]
```diff
@@ -203,7 +207,8 @@ def _handle_table_cell_changed(self, row, col):
                     self.plotted.discard(ws_name)
             elif col == 3:
                 # subtract bg col
-                if is_sub or self.view.get_item_checked(row, 2):  # this ensures the sub ws isn't made on load
+                self.model.update_bgsub_status(ws_name, is_sub)
+                if is_sub or is_plotted:  # this ensures the sub ws isn't made on load
```
Comment (DannyHindson) [inline scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_presenter.py:211]: Can I just check the "or" logic here. The comment says "ensure the sub ws isn't made on load" but this does seem to be what happens. The background subtracted ws is created on load.
If the logic is correct and this is an "or" then perhaps the comment needs updating?

**Review by DannyHindson — COMMENTED** [review]
I see the "Subtract BG" check box being set to True now when I load some runs into the Fitting tab. So seems like it's working OK. Just one question about whether the subtraction should be happening on load or not

File: scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_presenter.py (hunk 1 above)
Comment (RichardWaiteSTFC) [reply scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_presenter.py:211]: Yes I meant to remove the comment - thanks for spotting this!

**Review by DannyHindson — APPROVED** [review]
I don't think you'd got around to changing the comment but rather than have this sit around for another week (until I'm back from holiday) I'll approve now. We're doing enough work on the engineering diffraction UI that we can pick this up separately

**Pull request merged**